sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreReader.java

```diff
@@ -588,7 +600,9 @@ private Mono<ReadReplicaResult> readPrimaryInternalAsync(
                             this.readFromStoreAsync(
                                 primaryUri,
                                 entity);
-                        replicaStatusList.set(Arrays.asList(primaryUri.getHealthStatusDiagnosticString()));
+                        replicaStatuses.remove(primaryUri);
+                        replicaStatusList.put(Uri.NOT_ATTEMPTING, new ArrayList<>(replicaStatuses.values()));
```

Comment:
Dito - why new list and copy?

Reply:
I was doing it convert into a list because .values() returns a collection. Decided to change it to a set instead of using a map and a list. No copying over is happening now.